pkg/products/threescale/reconciler.go

```diff
@@ -1043,26 +1055,35 @@ func (r *Reconciler) reconcileComponents(ctx context.Context, serverClient k8scl
 			"threescale_component_element": "zync-que",
 		})
 
-		apicastProdResources := corev1.ResourceRequirements{
-			Requests: corev1.ResourceList{corev1.ResourceCPU: k8sresource.MustParse("300m"), corev1.ResourceMemory: k8sresource.MustParse("250Mi")},
-			Limits:   corev1.ResourceList{corev1.ResourceCPU: k8sresource.MustParse("600m"), corev1.ResourceMemory: k8sresource.MustParse("300Mi")},
-		}
-		apim.Spec.Apicast.ProductionSpec.Resources = &apicastProdResources
+		if r.installation.Spec.Type == string(integreatlyv1alpha1.InstallationTypeManaged) {
+			apicastProdResources := corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: k8sresource.MustParse("300m"), corev1.ResourceMemory: k8sresource.MustParse("250Mi")},
+				Limits:   corev1.ResourceList{corev1.ResourceCPU: k8sresource.MustParse("600m"), corev1.ResourceMemory: k8sresource.MustParse("300Mi")},
+			}
+			apim.Spec.Apicast.ProductionSpec.Resources = &apicastProdResources
+
+			backendWorkerResources := corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: k8sresource.MustParse("150m"), corev1.ResourceMemory: k8sresource.MustParse("100Mi")},
+				Limits:   corev1.ResourceList{corev1.ResourceCPU: k8sresource.MustParse("300m"), corev1.ResourceMemory: k8sresource.MustParse("100Mi")},
+			}
+			apim.Spec.Backend.WorkerSpec.Resources = &backendWorkerResources
 
-		backendWorkerResources := corev1.ResourceRequirements{
-			Requests: corev1.ResourceList{corev1.ResourceCPU: k8sresource.MustParse("150m"), corev1.ResourceMemory: k8sresource.MustParse("100Mi")},
-			Limits:   corev1.ResourceList{corev1.ResourceCPU: k8sresource.MustParse("300m"), corev1.ResourceMemory: k8sresource.MustParse("100Mi")},
+			backendListenerResources := corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: k8sresource.MustParse("250m"), corev1.ResourceMemory: k8sresource.MustParse("450Mi")},
+				Limits:   corev1.ResourceList{corev1.ResourceCPU: k8sresource.MustParse("600m"), corev1.ResourceMemory: k8sresource.MustParse("500Mi")},
+			}
+			apim.Spec.Backend.ListenerSpec.Resources = &backendListenerResources
 		}
-		apim.Spec.Backend.WorkerSpec.Resources = &backendWorkerResources
 
-		backendListenerResources := corev1.ResourceRequirements{
-			Requests: corev1.ResourceList{corev1.ResourceCPU: k8sresource.MustParse("250m"), corev1.ResourceMemory: k8sresource.MustParse("450Mi")},
-			Limits:   corev1.ResourceList{corev1.ResourceCPU: k8sresource.MustParse("600m"), corev1.ResourceMemory: k8sresource.MustParse("500Mi")},
+		if r.installation.Spec.Type == string(integreatlyv1alpha1.InstallationTypeManagedApi) {
+			err = productConfig.Configure(apim)
 		}
-		apim.Spec.Backend.ListenerSpec.Resources = &backendListenerResources
 
 		owner.AddIntegreatlyOwnerAnnotations(apim, r.installation)
 
+		if err != nil {
```

Comment:
should this error check be moved up to inside the if statement for the Configure function? Line 1079

Reply:
moved inside the if statement